blaze/expr/datetime.py

```diff
@@ -131,14 +131,14 @@ def utcfromtimestamp(expr):
 
 
 class nanosecond(DateTime): _dtype = datashape.int64
-class week(DateTime): _dtype = datashape.int16
+class week(DateTime): _dtype = datashape.int64
```

Comment:
This change makes the tests pass; however, https://github.com/blaze/blaze/pull/1585 intentionally made the datetime types as narrow as possible.

I ran across your proposed fix while looking at the following failure when running the test `py.test -s -x blaze/compute/tests/test_postgresql_compute.py::test_datetime_access[week]` against current master of `blaze`, `odo`, and `datashape`:

```
obj = 'Attributes', message = 'Attribute "dtype" are different', left = dtype('int16')
right = dtype('int64'), diff = None

    def raise_assert_detail(obj, message, left, right, diff=None):
        if isinstance(left, np.ndarray):
            left = pprint_thing(left)
        if isinstance(right, np.ndarray):
            right = pprint_thing(right)
    
        msg = """{0} are different
    
    {1}
    [left]:  {2}
    [right]: {3}""".format(obj, message, left, right)
    
        if diff is not None:
            msg = msg + "\n[diff]: {diff}".format(diff=diff)
    
>       raise AssertionError(msg)
E       AssertionError: Attributes are different
E       
E       Attribute "dtype" are different
E       [left]:  int16
E       [right]: int64

/mnt/envs/blaze/lib/python3.4/site-packages/pandas/util/testing.py:1018: AssertionError
```

(It might make sense to break off making the aforementioned test pass into a separate issue/PR.)

Reply:
This patch may checks the values, but ignores the dtypes.

```
$ git diff
diff --git a/blaze/compute/tests/test_postgresql_compute.py b/blaze/compute/tests/test_postgresql_compe.
py                                                                                                     
index 6c15703..608e8fb 100644
--- a/blaze/compute/tests/test_postgresql_compute.py
+++ b/blaze/compute/tests/test_postgresql_compute.py
@@ -886,4 +886,5 @@ def test_datetime_access(attr, sql_with_dts):
         compute(expr, sql_with_dts, return_type=pd.Series),
         getattr(compute(s.A, sql_with_dts, return_type=pd.Series).dt, attr),
         check_names=False,
+        check_dtype=False,
     )
```

`compute(s.A, sql_with_dts, return_type=pd.Series).dt` is an instance of `pandas.tseries.common.DatetimeProperties` which is going to return a `Series` with a dtype of `int64` when accessing the `week` attribute.

Reply:
https://github.com/blaze/blaze/pull/1660 applies the patch in the previous comment.